extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Specialized;
+using Microsoft.Azure.Functions.Worker.Core;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.Functions.Worker.Converters
+{
+    /// <summary>
+    /// Converter to bind Blob Storage type parameters.
+    /// </summary>
+    internal class BlobStorageConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (context.Source is not IBindingData bindingData)
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            if (bindingData.Source is not "Microsoft.Azure.WebJobs.Extensions.Storage.Blobs")
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            var content = bindingData.Content.ToObjectFromJson<Dictionary<string, string>>();
+
+            content.TryGetValue("Connection", out var connectionName);
+            content.TryGetValue("BlobName", out var blobName);
+            content.TryGetValue("ContainerName", out var containerName);
+
+            if (string.IsNullOrEmpty(connectionName)
+                || string.IsNullOrEmpty(blobName)
```

Comment:
You're always expecting a blob name, but blob extension also allows you to bind to a BlobContainerClient. Are you supporting that?

Reply:
Updated to have support for BlobContainerClient. I haven't actually tested any cases with IEnumerable or anything that might impact cardinality, will see if that works out of the box or requires more changes

Reply:
Will be done as part of this work - https://github.com/Azure/azure-functions-dotnet-worker/issues/1156